src/pages/studyView/StudyViewPageStore.ts

```diff
@@ -146,13 +145,63 @@ export class StudyViewPageStore {
         this._cnaGeneFilter = _cnaGeneFilter;
     }
 
-    @action hideChart(attributeId: string) {
-        this._attributeChartVisibility.set(attributeId, false);
+    @action changeChartVisibility(uniqueKey: string, visibility:boolean) {
+        if(!visibility){
+            //TODO: Currently clears only clinicalDataEqualityFilters,Need to implement for others
+            this._clinicalDataEqualityFilterSet.delete(uniqueKey);
+        }
+        this._chartVisibility.set(uniqueKey, visibility);
+    }
+
+    @action changeChartType(uniqueKey: string, chartType:ChartType, updateDefaultType?:boolean){
+        let chartMetaObject = this._clinicalAttributesMetaSet.get(uniqueKey);
+        if(chartMetaObject){
+            chartMetaObject.chartType = chartType;
+            if(updateDefaultType){
+                chartMetaObject.defaultChartType = chartType
+            }
+            this._clinicalAttributesMetaSet.set(uniqueKey, chartMetaObject);
+        }
+    }
+    @action onChangeSelectedClinicalChart(options: {label: string; value: string;}[]) {
+        this._chartVisibility.clear();
+        for (const option of options) {
+            this.changeChartVisibility(option.value, true)
+        }
+    }
+
+    @computed get clinicalChartOptions() {
+        return _.map(this.clinicalAttributesMeta.result, clinicalAttributeMeta=>({
+            label: clinicalAttributeMeta.clinicalAttribute.displayName,
+            value: clinicalAttributeMeta.uniqueKey
+        }));
+    }
+
+    @computed get selectedClinicalCharts() {
+        return _.reduce(this._chartVisibility.keys(),(acc:{label: string; value: string;}[], next)=>{
+            if(this._chartVisibility.get(next)) {
+                let chartMeta = this._clinicalAttributesMetaSet.get(next)
+                if(chartMeta){
+                    acc.push({
+                        label: chartMeta.clinicalAttribute.displayName,
+                        value: chartMeta.uniqueKey
+                    })
+                }
+            }
+            return acc;
+        },[]);
+    }
+
+    @computed get emptyFilters() {
+        let filters: StudyViewFilter = {} as any;
+        filters.studyIds = [this.studyId]
```

Comment:
will we sometimes have more than one studyId?  If so, should we engineer for that?  i.e. this.studyIds    instead of single studyId